cidlists/cidlists.go

```diff
@@ -0,0 +1,102 @@
+package cidlists
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ipfs/go-cid"
+	cbg "github.com/whyrusleeping/cbor-gen"
+
+	datatransfer "github.com/filecoin-project/go-data-transfer"
+)
+
+// CIDLists maintains files that contain a list of CIDs received for different data transfers
+type CIDLists interface {
+	CreateList(chid datatransfer.ChannelID, initalCids []cid.Cid) error
+	AppendList(chid datatransfer.ChannelID, c cid.Cid) error
+	ReadList(chid datatransfer.ChannelID) ([]cid.Cid, error)
+	DeleteList(chid datatransfer.ChannelID) error
+}
+
+type cidLists struct {
+	baseDir string
+}
+
+// NewCIDLists initializes a new set of cid lists in a given directory
+func NewCIDLists(baseDir string) (CIDLists, error) {
+	base := filepath.Clean(string(baseDir))
+	info, err := os.Stat(string(base))
+	if err != nil {
+		return nil, fmt.Errorf("error getting %s info: %s", base, err.Error())
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("%s is not a directory", base)
+	}
+	return &cidLists{
+		baseDir: base,
+	}, nil
+}
+
+// CreateList initializes a new CID list with the given initial cids (or can be empty) for a data transfer channel
+func (cl *cidLists) CreateList(chid datatransfer.ChannelID, initalCids []cid.Cid) error {
+	f, err := os.Create(transferFilename(cl.baseDir, chid))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = f.Close()
```

Comment:
May be worth returning the error on Close. It seems like the caller would want to know if the file couldn't be closed.